src/emulator/storage/apis/firebase.ts

```diff
@@ -128,6 +140,47 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
     return res.json(new OutgoingFirebaseMetadata(metadata));
   });
 
+  firebaseStorageAPI.post(`/b/:bucketId/o/:objectId[(:)]generateSignedUrl`, async (req, res) => {
+    let signedUrlObject: SignedUrlResponse;
+
+    try {
+      signedUrlObject = await storageLayer.generateSignedUrl({
+        bucketId: req.params.bucketId,
+        decodedObjectId: decodeURIComponent(req.params.objectId),
+        authorization: req.header("authorization"),
+        originalUrl: `${req.protocol}://${req.hostname}:${req.socket.localPort}`,
+        ttlSeconds:
+          req.body.ttlSeconds === undefined ? SIGNED_URL_DEFAULT_TTL_SECONDS : req.body.ttlSeconds,
+      });
+    } catch (err) {
+      if (err instanceof NotFoundError) {
+        return res.status(404).json({
+          error: {
+            code: 404,
+            message: `Object in bucket does not exist`,
```

Comment:
Could we have `generateSignedUrl` provide us the message? It could be more descriptive and include the object path.